tsfc/impero_utils.py

```diff
@@ -88,14 +88,10 @@ def collect_temporaries(ops):
         # IndexSum temporaries should be added either at Initialise or
         # at Accumulate.  The difference is only in ordering
         # (numbering).  We chose Accumulate here.
-        if isinstance(op, (imp.Initialise, imp.Return, imp.ReturnAccumulate)):
-            pass
-        elif isinstance(op, imp.Accumulate):
+        if isinstance(op, imp.Accumulate):
             result.append(op.indexsum)
         elif isinstance(op, imp.Evaluate):
             result.append(op.expression)
-        else:
-            raise AssertionError("unhandled operation: %s" % type(op))
```

Comment:
So previously you asserted (effectively) that you were passed on of `Initialise`, `Return`, `ReturnAccumulate`, `Accumulate` or `Evaluate`.  Now you silently pass through everything not explicitly handled.  I guess this is because now you have these `Noop` nodes too?